utils/author-precreate-profile.py

```diff
@@ -0,0 +1,124 @@
+###############################################################################
+# ex. python author-precreate-profile.py --baseurl http://localhost:3000
+#       --username admin --password admin_pw
+#
+# Find authors and precreate profiles if they are missing.
+# Possible to change out emails w/ tildeId's in future.
+###############################################################################
+
+## Import statements
+import argparse
+from openreview import *
+
+## Argument handling
+parser = argparse.ArgumentParser()
+parser.add_argument('--baseurl', help="base url")
+parser.add_argument('--username')
+parser.add_argument('--password')
+
+args = parser.parse_args()
+
+## Initialize the client library with username and password
+client = Client(baseurl=args.baseurl, username=args.username, password=args.password)
+
+"""
+Wrapper around OpenReview.get_profile to catch Profile not found exceptions.
+"""
+def get_profile(value):
+    profile = None
+    try:
+        profile = client.get_profile(value)
+    except openreview.OpenReviewException as e:
+        # throw an error if it is something other than "not found"
+        if e[0][0] != 'Profile not found':
+            print "OpenReviewException: {0}".format(e)
+            raise e
+    return profile
+
+"""
+Create user profiles -  the groups and profile are created, but not activated.
+The first time the user logs in they will have to fill in their information.
+"""
+def create_profile(email, first, last, tilde_id, verbose=False):
+
+    tilde_group = openreview.Group(tilde_id,
+                                   signatures = ["OpenReview.net"],
+                                   signatories=[tilde_id],
+                                   readers=[tilde_id],
+                                   writers=[tilde_id],
+                                   nonreaders=[],
+                                   members=[email])
+    if verbose: print "Generating new tilde group %s" % tilde_id
+    client.post_group(tilde_group)
+
+    email_group = openreview.Group(email,
+                                   signatures=["OpenReview.net"],
+                                   signatories=[email],
+                                   readers=[email],
+                                   writers=[email],
+                                   nonreaders=[],
+                                   members=[tilde_id])
+    client.post_group(email_group)
+
+    profile = openreview.Note.from_json({
+        'id': tilde_id,
+        'content': {
+            'emails': [email],
+            'preferred_email': email,
+            'names': [{
+                'first': first,
+                'middle': '',
+                'last': last,
+                'username': tilde_id
+            }]
+        }
+    })
+
+    # Why doesn't  "client.post_note(profile)" work? throws OpenReviewException: [u'note does not exist']
+    response = requests.put(client.baseurl+"/user/profile", json=profile.to_json(), headers=client.headers)
```

Comment:
We need a specific call to create profiles because we are defining the id outside of the call. POST notes set automatically a string id. This is failing because POST notes receive an id in the body and thinks that you are trying to edit a note, but the note doesn't exist. 

Obs: http PUT operations is used to create object and POST to update objects. We should split the POST notes in PUT and POST depending what you want to do.